tests/operator/operator_test.go

```diff
@@ -678,342 +667,7 @@ var _ = Describe("[Serial][sig-operator]Operator", Serial, decorators.SigOperato
-					return err
-				}, 10*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
-
-				// change run stategy to halted to be able to restore the vm
-				Eventually(func() bool {
-					vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(nil)).Get(vmYaml.vmName, &metav1.GetOptions{})
-					if err != nil {
-						return false
-					}
-					vm.Spec.RunStrategy = &runStrategyHalted
-					_, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Update(vm)
-					if err != nil {
-						return false
-					}
-					updatedVM, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Get(vmYaml.vmName, &metav1.GetOptions{})
-					Expect(err).ToNot(HaveOccurred())
-					return updatedVM.Spec.Running == nil && updatedVM.Spec.RunStrategy != nil && *updatedVM.Spec.RunStrategy == runStrategyHalted
-				}, 30*time.Second, 3*time.Second).Should(BeTrue())
-
-				By(fmt.Sprintf("Ensure vm %s can be restored from vmsnapshots", vmYaml.vmName))
-				for _, snapshot := range vmYaml.vmSnapshots {
-					_, _, err = clientcmd.RunCommand(k8sClient, "create", "-f", snapshot.restoreYamlFile, "--cache-dir", newClientCacheDir)
-					Expect(err).ToNot(HaveOccurred())
-					Eventually(func() bool {
-						r, err := virtClient.VirtualMachineRestore(testsuite.GetTestNamespace(nil)).Get(context.Background(), snapshot.restoreName, metav1.GetOptions{})
-						if err != nil {
-							return false
-						}
-						return r.Status != nil && r.Status.Complete != nil && *r.Status.Complete
-					}, 180*time.Second, 3*time.Second).Should(BeTrue())
-				}
-
-				By(fmt.Sprintf("Deleting VM with %s api", vmYaml.apiVersion))
-				_, _, err = clientcmd.RunCommand(k8sClient, "delete", "-f", vmYaml.yamlFile, "--cache-dir", newClientCacheDir)
-				Expect(err).ToNot(HaveOccurred())
-
-				By("Waiting for VM to be removed")
-				Eventually(func() bool {
-					_, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(nil)).Get(vmYaml.vmName, &metav1.GetOptions{})
-					if err != nil && errors.IsNotFound(err) {
-						return true
-					}
-					return false
-				}, 90*time.Second, 1*time.Second).Should(BeTrue())
-			}
-
-			By("Verifying all migratable vmi workloads are updated via live migration")
-			verifyVMIsUpdated(migratableVMIs, launcherSha)
-
-			if len(migratableVMIs) > 0 {
-				By("Verifying that a once migrated VMI after an update can be migrated again")
-				vmi := migratableVMIs[0]
-				migration, err := virtClient.VirtualMachineInstanceMigration(vmi.Namespace).Create(tests.NewRandomMigration(vmi.Name, vmi.Namespace), &metav1.CreateOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				Eventually(ThisMigration(migration), 180).Should(HaveSucceeded())
-			}
-
-			By("Deleting migratable VMIs")
-			deleteAllVMIs(migratableVMIs)
-
-			By("Deleting KubeVirt object")
-			deleteAllKvAndWait(false)
-		},
-			Entry("by patching KubeVirt CR", false),
-			Entry("by updating virt-operator", true),
-		)
-	})
+	updateTest()
```

Comment:
Why this approach? Can't we have 2 separated files (without the first one calling into the second one)

Reply:
My first attempt to split it totally resulted in not functioning tests. Reason being the dependencies between variables, before/after blocks and tests seems to be too much and therefore require a more iterative approach to this. I will iterate a few times over this with upcoming PRs and I think we will get there. Sounds good?

Reply:
Gotcha, no problem with that.